tests/unittests/RecommendationSystemTest.cpp

```diff
@@ -769,60 +821,37 @@ class RecommendationSystemTest : public BackendTest {
       }
     }
 
-    // Clear the EE to free memory up for other runs to compare against.
-    EE_.clear();
-
     // Compare against interpreter if we're not executing already on it.
     if (getBackendName() != "Interpreter") {
       compareAgainstInterpreter();
     }
   }
 
-  /// Execute a graph of functions based on the given DAG.
-  void executeDAG(ExecutionEngine &partitionedEE, DAGNode *G,
-                  ExecutionContext &context, CompilationContext &cctx) {
-    std::vector<DAGNode *> exeList;
-    int endPt = 0;
-    int curPt = 0;
-    // The first node is always the dummy node.
-    exeList.push_back(G);
-    endPt++;
-    // All the functions are in the module, compile them all at once.
-    partitionedEE.compile(cctx);
-
-    while (curPt < endPt) {
-      DAGNode *dag = exeList.at(curPt);
-      // The root in a G is always a dummy function.
-      if (curPt > 0) {
-        updateInputPlaceholders(*context.getPlaceholderBindings(), {}, {});
-        partitionedEE.run(context, dag->name);
-      }
-      for (unsigned int i = 0, e = dag->children.size(); i < e; i++) {
-        exeList.push_back(dag->children.at(i));
-        endPt++;
-      }
-      curPt++;
-    }
-  }
-
   /// Run on the Interpreter and compare the result to previous result.
   void compareAgainstInterpreter() {
     ExecutionContext contextI;
-    // Set device memory to 64GB to prevent partitioning. We are using the
-    // Interpreter's result just as a reference result to compare against.
-    ExecutionEngine IEE("Interpreter", 64e+9);
-    auto *modI = &IEE.getModule();
+
+    std::unique_ptr<Module> modI(new Module);
```

Comment:
Could you add some comments inside this function. Please explain some of the variables like  `modI` (interpreted module?), `pMod` (partitioned module?). 

Also please use a more uniform naming, e.g. use only (descriptive) prefixes or suffixes for indicating the meaning, but not both styles at the same time ;-)